proxy/coco/src/peer/subroutines.rs

```diff
@@ -0,0 +1,280 @@
+//! Management of subroutines.
+
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+use futures::stream::{BoxStream, FuturesUnordered, SelectAll, StreamExt as _};
+use tokio::{
+    sync::{broadcast, mpsc},
+    time::interval,
+};
+
+use librad::{
+    net::{peer::Gossip, protocol::ProtocolEvent},
+    peer::PeerId,
+    uri::{RadUrl, RadUrn},
+};
+
+use crate::{
+    convert::MaybeFrom as _,
+    spawn_abortable::{self, SpawnAbortable},
+    state::State,
+};
+
+use super::{
+    announcement, control, gossip,
+    run_state::{
+        AnnounceInput, Command, Config as RunConfig, ControlCommand, Event, Input, RequestCommand,
+        RequestInput, RunState, SyncInput, TimeoutInput,
+    },
+    sync, RECEIVER_CAPACITY,
+};
+
+/// Management of "subroutine" tasks.
+#[allow(clippy::missing_docs_in_private_items)]
+pub struct Subroutines {
+    pending_tasks: FuturesUnordered<SpawnAbortable<()>>,
+    inputs: SelectAll<BoxStream<'static, Input>>,
```

Comment:
Could you document these fields? What is `inputs` and input for? for `pending_tasks`? I'd say no but don't know